server/src/main/java/io/druid/client/HttpServerInventoryView.java

```diff
@@ -625,32 +605,80 @@ public void onFailure(Throwable t)
             },
             executor
         );
-
-        return future;
       }
       catch (Throwable th) {
```

Comment:
I guess retrying is not possible for all kinds of exceptions (ex, OOM). Better to specify more exact types of expected exceptions.

Reply:
here i would like to retry in all cases unless this server disappears from zookeeper not risk missing this server before its gone. this catch all is really for any kind of unexpected exception etc.
for OOM, user can always specify `-XX:+ExitOnOutOfMemory` to kill the jvm.

Reply:
The ExitOnOutOfMemoryError option can be a way to avoid retrying on OOM, but I can't find any documents about that in druid. So, users may not specify that option, and thus I think basically we need to avoid catching all types of exceptions especially unrecoverable fatal exceptions. Since this is not added in this pr, I'm fine with opening an issue for this.

Also, I think there will be a better way to synchronize the server list got from zk and http. What do you think about opening an issue for this?

Reply:
those options can be added to druid document, but also note that they are general jvm options and documented elsewhere too. We use catch Throwable in many difference places in Druid in critical situations where we don't want to miss any unknown situation. In the worst case if error is truly unrecoverable (and server does not disappear) beyond unstability timeout then it starts to send alerts also.
I'm sure there could be other ways to synchronize segment list but wouldn't want to do further work on it unless current way is broken or is significantly inferior to other approaches.
I created an issue to document os/jvm stuff though ( https://github.com/druid-io/druid/issues/4792 )

Reply:
Ok.